server/src/main/java/org/elasticsearch/action/search/SearchTask.java

```diff
@@ -53,4 +54,19 @@ public final SearchProgressListener getProgressListener() {
         return progressListener;
     }
 
+    /**
+     * @return the Supplier of {@link SearchResponseMerger} attached to this task. Will be null
+     * for local-only search and cross-cluster searches with minimize_roundtrips=false.
+     */
+    public Supplier<SearchResponseMerger> getSearchResponseMergerSupplier() {
```

Comment:
This is a bit odd. Not a huge deal, but perhaps we could look at whether it makes sense to provide what SearchTask needs to create the merger independently. COuld also be done as a follow-up.

Reply:
As I said in my other comment, this is one of the tricky bits of the design. The information needed to create a SearchResponseMerger is:

```
        SearchSourceBuilder source,
        SearchTimeProvider timeProvider,
        AggregationReduceContext.Builder aggReduceContextBuilder
```

None of which are available to the SearchTask and 2 of which are created in the TransportSearchAction, so you would have to create setters to apply all this information into the SearchTask. That seems like a code smell to me (law of Demeter violation). The Supplier seems much cleaner - basically a closure you can call on demand to create new ones as needed.